libshaderc/src/shaderc_cpp_test.cc

```diff
@@ -504,6 +505,106 @@ TEST_F(CppInterface, PreprocessingOnlyModeSecondOverridesDisassemblyMode) {
               HasSubstr("void main(){ }"));
 }
 
+// A mock class that simulate a client of the includer.
+class TestIncluder : public shaderc::CompileOptions::Includer {
+ public:
+  // Create a fake file object, with specified path and content.
+  struct FakeFile {
+    std::string path;
+    std::string content;
+  };
+  // Use hashmap as a fake file system to store fake files to be included.
+  using FakeFS = std::unordered_map<std::string, FakeFile>;
+
+  TestIncluder(FakeFS& fake_fs) : fake_fs_(fake_fs){};
```

Comment:
The semicolon is redunant.
Also, I suspect this isn't clang-formatted.    I expected a space before {}